dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Domain.xml

```diff
@@ -1,7 +1,8 @@
 <model>
     <mount>//OPNsense/bind/domain</mount>
     <description>BIND domain configuration</description>
-    <version>1.0.1</version>
+    <version>1.1.0</version>
+    <migration_prefix>Domain</migration_prefix>
```

Comment:
Doesn’t this require changing all other migration file names as well? What’s the idea here?

Reply:
I wanted to ensure that someone else bumping another one of the models to the same version number wouldn’t run this migration. I tested this and it seems to work without changing all the other migrations prefixes. I have all the code written to add full tsig key management and it has another migration for the domain model while also adding a model and those all seem to work with just the one prefix specified.

Reply:
I'm not convinced this will hold up with future migrations or on installs where migrations of old type need to be carried out as well from the perpsective of the latest model definition.

Reply:
Since this is the first migration for this plugin I think it works without issue. 

I can also re-implement with something like this if you feel that's safer: https://github.com/opnsense/plugins/blob/16f3522d08d30919b17e66bdec38352ef4c75208/net/frr/src/opnsense/mvc/app/models/OPNsense/Quagga/Migrations/M1_0_2.php#L43

General migrations question: Do you have an example of pulling a field out of a model and moving it into its own model? I have a working migration to move the TSIG keys out of the `Domains` model into it's own model, but the only way I can get that working wasn't the way I wanted to do it. I guess we can chat about it on that migration later if needed. Here is the code if interested https://github.com/sestary/opnsense-plugins/blob/keys/dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Migrations/Domain1_1_0.php. I haven't made a PR for this stuff yet since there are too many conflicting changes with this PR.

Reply:
> General migrations question: Do you have an example of pulling a field out of a model and moving it into its own model?

I don’t expect we do have an example as splitting functional models (as some plugins do) have the issue of causing possible race conditions (who is responsible for which change at every moment in time). My best guess at the moment would be to instantiate the new models from the migration of the existing model if there’s no data in the new on yet…. But this will likely fail in some scenario.

Our current migrations are either scoped within the same model or are migrating legacy components.

Most of our own models are build for a functional module instead of an entity. (entities within the same app share the same model)

Reply:
Thanks, we can discuss that more on the other migration once I create that PR. Let me know if you think it'll be better to re-implement the migration with the `instanceof` if condition as per the link above.

Reply:
The targeting issue can indeed be solved with an ‘instanceof’, one example for that case is https://github.com/opnsense/core/commit/7330d12282c5afe7e4b3157db34844b586b990ce

Which is likely a good idea anyway to prevent different models of frr executing the migration.

The concurrency problem (different models same functional area) will likely still cause issues at some point.

Reply:
I've updated it like in that example, I will use that model for the other migration as well.